src/components/common/RestEntity/README.md

```diff
@@ -0,0 +1,882 @@
+      {/*
+        okay... so what the hell did this do? 
+        Now the RestEntity has supposedly fetched my data.
+        But how do I access anything with it? 
+      */}
+      <RestEntity.Singleton />
+      <RestEntity.Error />
+      <RestEntity.Loading />
+    </RestEntity>
+  );
+};
+
+export default ProgramComponent;
+```
+
+Our `<RestEntity href=“/endpoint”>` parent component is going to take in that `href` prop and hit `GET /programs/program/${programId}` . Bang. Boom. In one line of code, you have effectively triggered a GET request to an endpoint. For now, that’s all you need to know about ***why*** that hits an endpoint: because you told it to.
+
+NOTE: I'll add information about how the RestEntity works under the hood later
+
+Okay, so what about the next part? How do I DO anything with that data? And how does the whole Loading & Error state come into play?
+
+That's where the sub-components come in.
+
+We have the `Singleton` for success state, `Error` for error state, and `Loading` for loading state.
+
+
+Let's think all the way back to the original `ProgramComponent.js`. What did its return statement look like?
+
+```javascript
+return (
+    <>
+    {
+      // if data exists and is truthy (i.e., not `undefined` or `null`)
+      // then we'll render our program card with information!
+      data ?
+        <div>
+          <h1>Program Name: {data.programName}</h1>
+          <p>Description: {data.description}</p>
+        </div>
+        // otherwise, we'll show the user that we're loading up their data
+        : <div>Loading...</div>
+    }
+    {
+      // if error exists, then we'll render this warning to our user!
+      error && <p>Oh no! There's been a problem!</p>
+    }
+    </p>
+  );
+```
+
+It was handling a bunch of logic to decide whether it was in a place to show data or not.
+
+But the essence behind all the ternaries and truthiness fail-safes contained three different
+views based on the three shapes of state (LOADING, SUCCESS, ERROR).
+
+LOADING:
+
+```javascript
+// if data is still being fetched
+// we'll show the user that we're loading up their data
+return <div>Loading...</div>;
+```
+
+ERROR:
+
+```javascript
+// if we hit an error, then we'll render this warning to our user!
+return <p>Oh no! There's been a problem!</p>;
+```
+
+SUCCESS:
+
+```javascript
+// if data was successfully fetched, 
+// then we'll render our program JSX to display that data
+return  (
+  <div>
+    <h1>Program Name: {data.programName}</h1>
+    <p>Description: {data.description}</p>
+  </div>
+);
+```
+
+Do you notice anything interesting there? Of the three different states, only ONE depends on external data of any sort—SUCCESS. We can display anything we want to for a Loading message or an Error message... the only thing that depends on the actual data is the success!! And logically so, because if you're loading data or failed to get the data, then you definitely don't have any data that you could depend on.
```

Comment:
What is unclear to me in these statements is whether this is only occurring for this scenario or whether it will occur in all scenarios. Perhaps those distinctions should be made clearer throughout the doc. What are the hard and fast rules for the rest entities and what will be up to the engineer.

Reply:
good point. 

If I had a separate section for hard and fast rules for Rest Entity, what’s possible vs what’s required... where should that go, section-wise??